packages/flutter/lib/src/widget_previews/widget_previews.dart

```diff
@@ -70,17 +75,14 @@ base class Preview {
   /// If not provided, no name will be associated with the preview.
   final String? name;
 
-  /// Artificial width constraint to be applied to the previewed widget.
+  /// Artificial constraints to be applied to the previewed widget.
   ///
-  /// If not provided, the previewed widget will attempt to set its own width
-  /// constraints and may result in an unbounded constraint error.
-  final double? width;
-
-  /// Artificial height constraint to be applied to the previewed widget.
+  /// If not provided, the previewed widget will attempt to set its own
+  /// constraints.
   ///
-  /// If not provided, the previewed widget will attempt to set its own height
-  /// constraints and may result in an unbounded constraint error.
-  final double? height;
+  /// If a dimension has a value of `double.infinity`, the previewed widget
+  /// will attempt to set its own constraints in the relevant dimension.
```

Comment:
To keep folks from explicitly setting infinity like you mentioned. :)

```suggestion
  /// will attempt to set its own constraints in the relevant dimension.
  ///
  /// To set a single dimension and allow the other to set its own constraints, use [Size.fromHeight] or [Size.fromWidth].
```

Reply:
Good call!